api/handlers/modalityhandler.py

```diff
@@ -0,0 +1,170 @@
+from ..web import base
+from .. import config
+from ..auth import require_login, require_admin
+from ..dao import containerstorage
+from ..web.errors import APINotFoundException, APIValidationException
+#from ..validators import validate_data
+
+log = config.log
+
+
+class ModalityHandler(base.RequestHandler):
+
+    def __init__(self, request=None, response=None):
+        super(ModalityHandler, self).__init__(request, response)
+        self.storage = containerstorage.ContainerStorage('modalities', use_object_id=False)
+
+    @require_login
+    def get(self, modality_name):
+        return self.storage.get_container(modality_name)
+
+    @require_login
+    def get_all(self):
+        return self.storage.get_all_el(None, None, None)
+
+    @require_admin
+    def post(self):
+        payload = self.request.json_body
+        # Clean this up when validate_data method is fixed to use new schemas
```

Comment:
Does this still apply? Why can't `validate_data` use the newly added `modality.json`?

Reply:
This is also a TODO that was intended to be finished before PR review. I ended up using a lot of TODO-type comments since this branch existed for so long.